tests/components/qwikswitch/test_init.py

```diff
@@ -98,18 +67,21 @@ def aioclient_mock():
             }
         }
     }
+    aioclient_mock.get("http://127.0.0.1:2020/&device", text=DEVICES)
+    aioclient_mock.get("http://127.0.0.1:2020/&listen", long_poll=True)
     await async_setup_component(hass, QWIKSWITCH, config)
-
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
     await hass.async_block_till_done()
+
     state_obj = hass.states.get("sensor.ss1")
     assert state_obj.state == "None"
 
-    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
-
-    LISTEN.append(
-        '{"id":"@a00001","name":"ss1","type":"rel",' '"val":"4733800001a00000"}'
+    aioclient_mock.get(
+        "http://127.0.0.1:2020/&listen",
+        long_poll=True,
+        text='{"id":"@a00001","name":"ss1","type":"rel",' '"val":"4733800001a00000"}',
     )
-
+    await asyncio.sleep(0.01)
```

Comment:
Maybe call a `aioclient_mock.stop()` here, just to test the timeout (and sleep) mechanism in the library? Then you can remove this sleep

Reply:
or we can push a exc=ClientError()?

Reply:
the problem is that there are multiple commands. If i call stop after one, the next ones won't happen
Since this is all going asyncronously on the event loop, i couldn't think of a better mechanism than a short sleep
i would suggest adding a test for ClientError(), so the tests simulate the behavior we want to (e.g. that a client error disconnects)

Reply:
The sleep is a good idea, but since this is the last time we read from the mock we can use either stop() or a ClientError()

For your info, in the library a ClientError() return from a aiohttp.get will result in a asyncio.sleep inside the library - [see here](https://github.com/kellerza/pyqwikswitch/blob/master/pyqwikswitch/async_.py#L91)

Reply:
i know. i had to look in the library code to understand it. however, i believe it is better to separate the internal library implementation from the unit test that should check behaviors. if tomorrow the internal library behavior changes, we would prefer that it wouldn't break the test as long as the library still does what it should

Reply:
👍  I should probably add some unit testing to the library itself

Reply:
that is what i did on the integration i was writing (dynalite). The integration assumes the library is a black box and the library has its own tests

Reply:
Do we need to sleep for some time or is it enough that we yield once to the event loop? If it's just the latter we can sleep for 0 seconds.
```py
await asyncio.sleep(0)
```

Reply:
the library does some of these things asynchronously. I am not sure if a single sleep(0) would cover it, since if there is a task that runs create_task for another task, it would fail.
I can try if with sleep(0) but worried that it won't be deterministic and we may have intermittent failures, which i prefer not to.
as far as i know there isn't anything that is not immediate, or in other words, the event loop will never be idle, but not sure if there is something to wait until event loop is idle (all tasks done or waiting on something) in python

Reply:
Its only in error conditions that another task is created, but the small wait is probably more deterministic, so I’m ok with that